swaylock/render.c

```diff
@@ -0,0 +1,146 @@
+#define _POSIX_C_SOURCE 199506L
+#include <math.h>
+#include <stdlib.h>
+#include <wayland-client.h>
+#include "cairo.h"
+#include "background-image.h"
+#include "swaylock/swaylock.h"
+
+#define M_PI 3.14159265358979323846
+
+void render_frame(struct swaylock_surface *surface) {
+	struct swaylock_state *state = surface->state;
+	surface->current_buffer = get_next_buffer(state->shm,
+			surface->buffers,
+			surface->width * surface->scale,
+			surface->height * surface->scale);
+	cairo_t *cairo = surface->current_buffer->cairo;
+	cairo_identity_matrix(cairo);
+
+	int buffer_width = surface->width * surface->scale;
+	int buffer_height = surface->height * surface->scale;
+	if (state->args.mode == BACKGROUND_MODE_SOLID_COLOR) {
+		cairo_set_source_u32(cairo, state->args.color);
+		cairo_paint(cairo);
+	} else {
+		render_background_image(cairo, surface->image,
+				state->args.mode, buffer_width, buffer_height);
+	}
+	cairo_identity_matrix(cairo);
+
+	int ARC_RADIUS = 50 * surface->scale;
```

Comment:
Why are these in all caps?

Reply:
Used to be constants. Will lowercase them now.